lib/HALTarget/IMU.h

```diff
@@ -0,0 +1,197 @@
+ * Includes
+ *****************************************************************************/
+#include "IIMU.h"
+#include "Zumo32U4IMU.h"
+
+/******************************************************************************
+ * Macros
+ *****************************************************************************/
+
+/******************************************************************************
+ * Types and Classes
+ *****************************************************************************/
+  
+/** The IMU adapter. */
+class IMU : public IIMU
+{
+public:
+    /**
+     * Constructs the IMU adapter.
+     *
+     */
+    IMU() :
+        IIMU(),
+        m_rawAccelerometerOffsetX(0),
+        m_rawAccelerometerOffsetY(0),
+        m_rawAccelerometerOffsetZ(0),
+        m_rawGyroOffsetX(0),
+        m_rawGyroOffsetY(0),
+        m_rawGyroOffsetZ(0)
+    {
+    }
+
+    /**
+     * Destroys the IMU adapter.
+     */
+    ~IMU()
+    {
+    }
+
+    template <typename T> struct vector
+    {
+        T x, y, z;
+    };
+
+    /**
+     * Initializes the inertial sensors and detects their type.
+     *
+     * @return True if the sensor type was detected succesfully; false otherwise.
+     */
+    bool init();
+        
+    /**
+     * Enables all of the inertial sensors with a default configuration.
+     */
+    void enableDefault();
+
+    /**
+     * 	Configures the sensors with settings optimized for turn sensing.
+     */
+    void configureForTurnSensing();
+
+    /**
+     * Takes a reading from the accelerometer and makes the measurements available in a.
+     */ 
+    void readAcc();
+        
+    /**
+     * Takes a reading from the gyro and makes the measurements available in g.
+     */
+    void readGyro();
+        
+    /**
+     * Takes a reading from the magnetometer and makes the measurements available in m.
+     */
+    void readMag();
+            
+    /**
+     * Indicates whether the accelerometer has new measurement data ready.
+     * 
+     * @return True if there is new accelerometer data available; false otherwise. 
+     */
+    bool accDataReady();
+        
+    /**
+     * Indicates whether the gyro  has new measurement data ready.
+     * 
+     * @return True if there is new gyro data available; false otherwise.
+     */
+    bool gyroDataReady();
+        
+    /**
+     * Indicates whether the magnetometer  has new measurement data ready.
+     * 
+     * @return True if there is new magnetometer data available; false otherwise.
+     */
+    bool magDataReady();
+
+    /**
+     * Get last raw Accelerometer values.
+     *
+     * @param[in] accelerationValues  Pointer to array where the Accelerometer values shall be written into.
+     */
+    void getAccelerationValues(int16_t* accelerationValues);
+
+    /**
+     * Get last raw Gyroscope values.
+     *
+     * @param[in] turnRates  Pointer to array where the Gyroscope values shall be written into.
+     */
+    void getTurnRates(int16_t* turnRates);
```

Comment:
Consider alternative way to declare array params (see previous comments).

Reply:
Leave it as a pointer since it fits the coding style of the rest of the Application (see e.g. RemoteControl/App.cpp/App_motorSpeedsChannelCallback)